### DISCUSSION TITLE
Buckets in a multi-tenant application

### DISCUSSION THREAD

**Discussion by annaisdevk** [supabase/supabase#16716] [Questions] [answered]
What is the best way to handle buckets in a multi-tenant application? Each tenant (organization) has many users.  And I am using Custom Claims to store tenant_id in auth.users table.

I had thought to create a new bucket for each tenant (the name of the bucket will be tenant_id) and configure policies(i.e. only members of this organization can perform CRUD in this bucket). 

Is there any kind of limitation?

**Comment by GaryAustin1** [comment, marked as answer]
There was just this discussion in Supabase Discord:  https://discord.com/channels/839993398554656828/1142871884925522001
